app/knight.py

```diff
@@ -0,0 +1,26 @@
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+    def armour_defence(self, armour: list) -> int:
+        for part in armour:
+            self.protection += part["protection"]
+        return self.protection
+
+    def use_of_weapons(self, weapon: dict) -> int:
+        self.power += weapon["power"]
+        return self.power
+
+    def apply_potion(self, potion: dict) -> None:
+        if potion:
+            potion_effect = potion["effect"]
+            for key in potion_effect:
+                if key == "hp":
+                    self.hp += potion_effect["hp"]
+                if key == "power":
+                    self.power += potion_effect["power"]
+                if key == "protection":
+                    self.protection += potion_effect["protection"]
```

Comment:
use `setattr` and `getattr`. If someone wants to add 5 more attributes. You must change your code